### PR TITLE
Added device ID override for VIVE wand support!

### PR DESCRIPTION
**Pull request by lucas-vrtech**

### CODE REVIEW

File: LucidGloves/src/DeviceDriver/KnuckleDriver.cpp [hunk 1]
```diff
@@ -188,7 +188,7 @@ void KnuckleDeviceDriver::StartDevice() {
 				vr::VRDriverInput()->UpdateBooleanComponent(m_inputComponentHandles[ComponentIndex::THUMBSTICK_TOUCH], datas.joyButton, 0);
 
 				vr::VRDriverInput()->UpdateBooleanComponent(m_inputComponentHandles[ComponentIndex::TRIGGER_CLICK], datas.trgButton, 0);
-				vr::VRDriverInput()->UpdateScalarComponent(m_inputComponentHandles[ComponentIndex::TRIGGER_VALUE], datas.trgButton, 0);
+				vr::VRDriverInput()->UpdateScalarComponent(m_inputComponentHandles[ComponentIndex::TRIGGER_VALUE], datas.flexion[1], 0);
```
Comment (danwillm) [inline LucidGloves/src/DeviceDriver/KnuckleDriver.cpp:191]: probably best to have this in a separate PR

File: LucidGloves/src/DeviceDriver/LucidGloveDriver.cpp [hunk 2]
```diff
@@ -57,7 +57,7 @@ vr::EVRInitError LucidGloveDeviceDriver::Activate(uint32_t unObjectId) {
 	const bool isRightHand = IsRightHand();
 
 	m_driverId = unObjectId; //unique ID for your driver
-	m_controllerPose = std::make_unique<ControllerPose>(m_configuration.role, std::string(lucidGlove::c_deviceManufacturer), m_configuration.offsetVector, m_configuration.angleOffsetVector, m_configuration.controllerIdOverride, m_driverId);
+	m_controllerPose = std::make_unique<ControllerPose>(m_configuration.role, std::string(lucidGlove::c_deviceManufacturer), m_configuration.offsetVector, m_configuration.angleOffsetVector, m_configuration.controllerIdOverride, m_configuration.isControllerOverride, m_driverId);
```
Comment (danwillm) [inline LucidGloves/src/DeviceDriver/LucidGloveDriver.cpp:60]: perhaps we can just provide the configuration to ControllerPose as well by casting to a shared pointer

We probably need to have a rethink to how we pass configuration, seems a little messy (though thats my fault)

**Review by danwillm — CHANGES_REQUESTED** [review]

File: LucidGloves/src/DeviceDriver/KnuckleDriver.cpp (hunk 1 above)
Comment (lucas-vrtech) [reply LucidGloves/src/DeviceDriver/KnuckleDriver.cpp:191]: Oops, I forgot to revert that

File: LucidGloves/src/DeviceDriver/LucidGloveDriver.cpp (hunk 2 above)
Comment (lucas-vrtech) [reply LucidGloves/src/DeviceDriver/LucidGloveDriver.cpp:60]: Yeah, let's do that in a separate PR

**Review by danwillm — APPROVED** [review]

**Pull request merged**